src/types/User.tsx

```diff
@@ -0,0 +1,80 @@
+export interface User {
+  id: number,
+  name: string;
+  username: string;
+  email: string;
+}
+
+// export const App = () => {
+//   const [goods, setGoods] = useState<Good[]>(goodsWithColor);
+//   const [newGoodName, setNewGoodName] = useState('');
+//   const [selectedColorId, setSelectedColorId] = useState(0);
+//   const [hasNameError, setHasNameError] = useState(false);
+//   const [hasColorError, setHasColorError] = useState(false);
+
+//   const addNewGood = (name: string, colorId: number) => {
+//     const newGood: Good = {
+//       id: Date.now(),
+//       name,
+//       colorId,
+//       color: getColorById(colorId),
+//     };
+
+//     setGoods((previousGoods) => [...previousGoods, newGood]);
+//   };
+
+//   const handleFormSubmit = (event: React.FormEvent) => {
+//     event.preventDefault();
+
+//     setHasNameError(!newGoodName);
+//     setHasColorError(!selectedColorId);
+
+//     if (newGoodName && selectedColorId) {
+//       addNewGood(newGoodName, selectedColorId);
+//     }
+//   };
+
+//   return (
+//     <div className="App">
+//       <form onSubmit={handleFormSubmit}>
+//         <input
+//           type="text"
+//           value={newGoodName}
+//           onChange={event => {
+//             setNewGoodName(event.target.value);
+//             setHasNameError(false);
+//           }}
+//           className={classNames({ 'with-error': hasNameError })}
+//         />
+//         {hasNameError && (
```

Comment:
Consider using a more descriptive error message. For example, `Name is required` could be `Please enter a name`. 📝